src/simulator_bridge.cpp

```diff
@@ -81,6 +92,90 @@ void modelStatesCallback(const gazebo_msgs::ModelStates& msg)
             depth_msg.header.stamp = ros::Time::now();
             depth_pub.publish(depth_msg);
         }
+
+        /*
+         * If the pinger index isn't found, break out and dont perform
+         * hydrophone calculations.
+         */
+        if (pinger_index > -1)
+        {
+            /*
+             * Once the submarine position and orientation are known, calculate
+             * all of the hydrophone positions. Assume the first hydrophone is
+             * 0.5 meters directly in front of the submarine, and that the 3
+             * other reference hydrophones are 0.3 meters along the primary
+             * axes. Begin by setting the hydrophone positions as relative to
+             * the submarine without any rotations applied. They are stored in
+             * order:
+             *      reference, x-axis, y-axis, z-axis
+             */
+            hydrophone_positions.push_back(Vector3d(0.5, 0, 0));
+            hydrophone_positions.push_back(Vector3d(0.8, 0, 0));
+            hydrophone_positions.push_back(Vector3d(0.5, 0.3, 0));
+            hydrophone_positions.push_back(Vector3d(0.5, 0, 0.3));
+
+            /*
+             * Next, create a yaw, pitch, and roll rotation matrix to calculate
+             * the rotated positions relative to the sub. Eigen can represent a
+             * rotation matrix as a quaternion. Then, once they are rotated,
+             * translate them to be with respect to the global frame by adding
+             * in the x, y, and z positions of the submarine.
+             */
+            Eigen::Quaterniond q(orientation_msg.quaternion.w,
```

Comment:
This step could be made unnecessary if you add each hydrophone as a separate link in the cobalt model and then just get their positions directly from the model states msg.